src/deploy/GoogleDeployer.js

```diff
@@ -32,25 +32,20 @@ class GoogleDeployer extends BaseDeployer {
   }
 
   ready() {
-    if (!this._client || !this._cfg.projectID || !this._cfg.keyFile) {
-      return false;
-    }
-    this._keyFilename = path.resolve(process.cwd(), this._cfg.keyFile);
-    // todo: maybe support async `ready` ?
-    if (fs.existsSync(this._keyFilename)) {
-      return true;
-    }
-    this.log.warn(`google client and project ID are specified, but key file at ${this._keyFilename} doesn't exist.`);
-    return false;
+    return !!this._client && !!this._cfg.projectID;
   }
 
   validate() {
     if (!this.ready()) {
       throw new Error('Google target needs key file, email, and project ID');
     }
+    if (!fs.existsSync(this._keyFilename)) {
+      throw new Error(`Google target needs key file ${this._keyFilename} which does not exist`);
```

Comment:
but the problem is that the CI tests provide the project ID, but not the keyfile. which should be a pre-requisite to auto-enable the deployer.

with this change, the auto-deploy to google will be enabled again, but then fails when the keyfile is missing, this should not be the case.

Reply:
The problem is that the CI tests provide the path to the keyfile, but this file isn't there. In 8927a79 I've moved the check to `ready()` so that the deployer will be disqualified from the auto-deploy rotation if the preconditions are not met.